react/components/RenderProvider.tsx

```diff
@@ -782,28 +795,45 @@ class RenderProvider extends Component<Props, RenderProviderState> {
     })
   }
 
-  public updateExtension = (name: string, extension: Extension) => {
+  public updateExtension = async (name: string, extension: Extension) => {
     const { extensions } = this.state
 
-    this.setState(
-      {
-        extensions: {
-          ...extensions,
-          [name]: extension,
+    return new Promise<void>(resolve => {
+      this.setState(
+        {
+          extensions: {
+            ...extensions,
+            [name]: extension,
+          },
         },
-      },
-      () => {
-        if (name !== 'store/__overlay') {
-          this.sendInfoFromIframe()
+        async () => {
+          if (name !== 'store/__overlay') {
+            await this.sendInfoFromIframe()
+          }
+
+          resolve()
         }
-      }
-    )
+      )
+    })
   }
 
   public handleSetDevice = (device: ConfigurationDevice) => {
     this.setState({ device })
   }
 
+  public addMessages = (newMessages: RenderRuntime['messages']) => {
```

Comment:
Should this be sync regardless triggering a promise? Don't we care about this promise result?

Reply:
Not now, but we may soon. I've made it async just in case.